### ISSUE TITLE
Dynamic Unit Map for units greater than 9

### ISSUE THREAD

**Issue by bbtest** [Tweaks/Tweaks#7]
If you add in a dynamic unit map and use a table with more than 9 rows at the left, the script gets confused and adds the dynamic link to the wrong row. For example, I have a dynamic unit map tweak in place and a table with a "Week" column down the left, numbered from 1 to 13. In week 13 there is an item I would like to link in the "Assessment" column. I have placed an item below the dynamic unit map and named it: "13: Assessment Essay 2". The script sees the "3:" in "13:" and thinks this item is supposed to be placed into Week 3's row. The script needs to be programmed in such a way that it looks for any digits preceding the digit it has found before the colon before adding it to the table.

**Comment by Tweaks** [comment]
This is a tricky one.. the default functionality of Tweaks allows people to use loose naming of items, allowing for flexibility in the link matching.. used in a bunch of sites. Let's see if we can propose a solution that doesn't break existing users sites.

Immediate workaround: Put a "W" or "Week " or any prefix in front of your numbers will make the strings unique.

Proposed solution:
Will update with a new regular expression to trim leading space from folder names and then search from the start of the item name.. so then will distinguish betweek 3: Lecture and 13: Lecture.
Does this sound ok? ETA prob about 10 days.

Code:
94: headers.filter(":contains('"+sectionTitle+": "+resourceTypes[column]+"')")
94: headers.filter(":startsWith('"+sectionTitle+": "+resourceTypes[column]+"')")

**Comment by bbtest** [comment]
Thanks for the prompt response. Sounds like your proposed solution would work. For now, I've implemented your suggested workaround by putting "Week " in front of the numbers in the "Week" column as well as in front of the items I'd like placed in the Dynamic Unit Map's table and that seems to work.

**Comment by Tweaks** [comment]
Hi, fix has been added for testing. Would you mind checking on your new and problem content? Will also ask my team back in Aus. You can test from the server here:
http://tweaks.github.com/Tweaks/description.html#unitmap
click 'toggle trial code' at the base of the page and embed this in your site instead of existing Tweak. If you were using this version already / you might have to clear cache. (help: http://tweaks.github.com/Tweaks/doco_trial.html ) thanks

**Comment by bbtest** [comment]
Thanks for the update. I tried the fix using the Remote Tweak method but I could not get it to work. I thought it might have been something to do with the security on our server and running remote scripts, so I also tested the scripts and CSS by placing them all in our Content Collection and trying from there, but still no luck. There are no errors coming up in the Error Console of Firefox and the script does seem to load as when you view it in Display View, the Remote Tweak item gets removed, however, the Unit Map table is not getting populated or styled. We are using Blackboard version 8 still, so that might be the problem.
